cdap-watchdog/src/main/java/io/cdap/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -210,7 +231,8 @@ public void stop() {
       LOG.info("Stopping appender {}", this.name);
       locationManager.close();
       if (encoder != null) {
```

Comment:
Is it just `super.close()`?  Also you are closing locationManager before that. Is the`OutputStream` still good here?

Reply:
BTW: `io.cdap.cdap.logging.plugins.LocationManager#close` seems to have a bug and does nothing. 
```
    Collection<LocationOutputStream> locations = activeLocations.values();
    activeLocations.clear();
```
`.values()` does not make a copy, so clear will make `locations` empty as well

Reply:
So, I feel that this writing footer should be done for all locations in the manager, not only for the current one. I am wondering if manager should take as `footerWriter` consumer or something